plugins/csv/src/lib.rs

```diff
@@ -79,3 +79,50 @@ impl Default for Config {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+
+    use crate::{Config, CsvPlugin};
+    use alumet::{
+        agent::{
+            self,
+            plugin::{PluginInfo, PluginSet},
+        },
+        pipeline,
+        plugin::PluginMetadata,
+    };
+    use std::time::Duration;
+
+    fn config_to_toml_table(config: &Config) -> toml::Table {
+        toml::Value::try_from(config).unwrap().as_table().unwrap().clone()
+    }
+
+    #[test]
+    fn start_stop() {
+        let default_config = Config { ..Default::default() };
+
+        let mut plugins = PluginSet::new();
+        plugins.add_plugin(PluginInfo {
+            metadata: PluginMetadata::from_static::<CsvPlugin>(),
+            enabled: true,
+            config: Some(config_to_toml_table(&default_config)),
+        });
+
+        // Set up the measurement pipeline
+        let mut pipeline = pipeline::Builder::new();
+        pipeline.normal_threads(2); // Example setting: use 2 threads to run async pipeline elements
+
+        // Build and start the agent
+        let agent = agent::Builder::from_pipeline(plugins, pipeline)
+            .build_and_start()
+            .expect("startup failure");
+
+        let handle = agent.pipeline.control_handle();
+        // Force shutdown
+        handle.shutdown();
+        agent
+            .wait_for_shutdown(Duration::new(5, 0))
+            .expect("error while running");
```

Comment:
Calling shutdown just after `build_and_start` and just before `wait_for_shutdown` is usually not a good idea: the pipeline will not run.